hazelcast/src/main/java/com/hazelcast/sql/impl/SqlServiceImpl.java

```diff
@@ -156,84 +164,106 @@ private SqlCursor query0(String sql, List<Object> params, long timeout, int page
         }
 
         if (timeout < 0) {
-            throw QueryException.error("Timeout cannot be negative: " + pageSize);
+            throw QueryException.error("Timeout cannot be negative: " + timeout);
         }
 
         if (pageSize <= 0) {
             throw QueryException.error("Page size must be positive: " + pageSize);
         }
 
         // Execute.
-        QueryState state;
-
         if (QueryUtils.isExplain(sql)) {
             String unwrappedSql = QueryUtils.unwrapExplain(sql);
 
             if (unwrappedSql.isEmpty()) {
                 throw QueryException.error("SQL statement to be explained cannot be empty");
             }
 
-            Plan plan = prepare(unwrappedSql);
+            DqlStatement operation = prepareQuery(unwrappedSql);
+            return operation.explain(internalService);
+        } else {
+            DqlStatement operation = prepareQuery(sql);
+            return operation.execute(internalService, params0, timeout, pageSize);
+        }
+    }
 
-            state = internalService.executeExplain(plan);
+    private DqlStatement prepareQuery(String sql) {
```

Comment:
It is not clear why SELECT is wrapped into a DDL class

Reply:
I'm not sure I follow.